docs/language/aggregates/and.md

```diff
@@ -18,7 +18,7 @@ echo 'true false true' | zq -z 'and(this)' -
 ```
 =>
 ```mdtest-output
-{and:false}
+false
```

Comment:
This PR can effectively be seen as one big bug fix. I see now that all this time the usage guidance for all our aggregate functions have looked like:

```
and(bool) -> bool
```

i.e., they've been showing the primitive return type this PR introduces and not the record type they've always returned. So now it finally matches. 😂

Reply:
> Whatever we ultimately decide, somewhere in the docs (probably in `docs/language/operators/summarize.md`) I think we should have some brief-but-explicit text explaining when they can expect what's returned to be a value vs. a record.

Ok, just pushed an update to summarize.md